src/serialization.jl

```diff
@@ -209,8 +209,8 @@ function deserialize(::Type{T}, data::Dict) where {T <: NamedTuple}
     return T(key = data[string(key)] for key in fieldnames(T))
 end
 
-# Some types that definitely won't be deserialized from Dicts
-const _NOT_FROM_DICT = Union{Nothing, Real, AbstractString}
+# Some types that definitely won't be deserialized from raw Dicts
+const _NOT_FROM_DICT = Union{Nothing, Real, AbstractString, TimeSeriesKey}
```

Comment:
I'm puzzled by this change. Can you explain?

Reply:
The deserialization code first checks to see if the dictionary to be deserialized contains concrete type information. If it does, it will deserialize to that directly and we'll never get to this union method. `TimeSeriesKey`s are always serialized with time series information (as are most structs, this list could be much longer) and therefore they will never be the answer when we get to this method.